ci/azure/pipelines.yml

```diff
@@ -21,30 +22,135 @@ jobs:
+      & 'C:\Program Files\7-Zip\7z.exe' x "${LLVM_CLANG_LLD_DIR}.tar.xz"
+      & 'C:\Program Files\7-Zip\7z.exe' x "${LLVM_CLANG_LLD_DIR}.tar"
+    name: install
+    displayName: 'Install LLVM/CLANG/LLD'
+
+  - pwsh: |
+      Set-Variable -Name ZIGBUILDDIR -Value "$(Get-Location)\build"
+      Set-Variable -Name ZIGINSTALLDIR -Value "$ZIGBUILDDIR\dist"
+      Set-Variable -Name ZIGPREFIXPATH -Value "$(Get-Location)\$(LLVM_CLANG_LLD_DIR)"
+
+      # Make the `zig version` number consistent.
+      # This will affect the cmake command below.
+      git config core.abbrev 9
+      git fetch --tags
+
+      mkdir $ZIGBUILDDIR
+      cd $ZIGBUILDDIR
+      & "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvarsall.bat" x64
+      cmake .. `
+        -Thost=x64 `
+        -G "Visual Studio 16 2019" `
+        -A x64 `
+        -DCMAKE_INSTALL_PREFIX="$ZIGINSTALLDIR" `
+        -DCMAKE_PREFIX_PATH="$ZIGPREFIXPATH" `
+        -DCMAKE_BUILD_TYPE=Release `
+        -DZIG_OMIT_STAGE2=ON 2> out-null
+      & "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\Bin\MSBuild.exe" `
+        /maxcpucount /p:Configuration=Release INSTALL.vcxproj
+    name: build
+    displayName: 'Build'
+
+  - pwsh: |
+      Set-Variable -Name ZIGINSTALLDIR -Value "$(Get-Location)\build\dist"
+
+      # Sadly, stage2 is omitted from this build to save memory on the CI server. Once self-hosted is
+      # built with itself and does not gobble as much memory, we can enable these tests.
+      #& "$ZIGINSTALLDIR\bin\zig.exe" test "..\test\behavior.zig" -fno-stage1 -fLLVM -I "..\test" 2>&1
+
+      & "$ZIGINSTALLDIR\bin\zig.exe" build test-toolchain -Dskip-non-native -Dskip-stage2-tests 2>&1
+      & "$ZIGINSTALLDIR\bin\zig.exe" build test-std -Dskip-non-native  2>&1
+    name: test
+    displayName: 'Test'
+
+  - pwsh: |
+      Set-Variable -Name ZIGINSTALLDIR -Value "$(Get-Location)\build\dist"
+
+      & "$ZIGINSTALLDIR\bin\zig.exe" build docs
+    timeoutInMinutes: 60
+    name: doc
+    displayName: 'Documentation'
+
   - task: DownloadSecureFile@1
     inputs:
+      name: s3cfg
       secureFile: s3cfg
-  - script: ci/azure/windows_msvc_script.bat
-    name: main
-    displayName: 'Build and test'
+
+  - pwsh: |
+      Set-Variable -Name ZIGBUILDDIR -Value "$(Get-Location)\build"
+      Set-Variable -Name AWSCFG -Value "$env:USERPROFILE\.aws\credentials"
+
+      # convert s3cfg to aws credentials
+      mkdir $(Split-Path -Path $AWSCFG)
+      echo "[default]" > $AWSCFG
+      (Select-String -Path "$(s3cfg.secureFilePath)" -Pattern "^access_key =" | `
+      select-object -ExpandProperty Line) -replace '^access_key', 'aws_access_key_id' | `
+      out-file -Append $AWSCFG
+      (Select-String -Path "$(s3cfg.secureFilePath)" -Pattern "^secret_key =" | `
+      select-object -ExpandProperty Line) -replace '^secret_key', 'aws_secret_access_key' | `
+      out-file -Append $AWSCFG
```

Comment:
I'm a little concerned about this hack here. Can you just show me the file path and example contents that need to be present and I will adjust the secret file to match it?

Reply:
It's indeed an ugly hack, it would be simpler to add a `crendentials` file.
The File path should be `C:\Users\{username}\.aws\credentials` so `$ $ENV:UserProfile\.aws\credentials` should work.
```
[default]
aws_access_key_id = my_access_key
aws_secret_access_key = my_secret_key
```

Reply:
I will make it so.

Reply:
There is now a secure file called `aws_credentials` in the format that you have specified. Can you update the push code accordingly?

Reply:
`s3cfg` replaced by `aws_credentials`

Reply:
The build is waiting for permission to access the new file, [message:](https://dev.azure.com/ziglang/zig/_build/results?buildId=19822&view=results)
> This pipeline needs permission to access a resource before this run can continue